File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let lastWord = words[0].charAt(0),
        idx = 0;
    const from = [];

    for (const word of words) {
        if (word.charAt(0) !== lastWord || from[word])
            return [(idx % n) + 1, Math.floor(idx / n) + 1];

        lastWord = word.slice(-1);
        ++idx;
        from[word] = true;
    }

    return [0, 0];
}

File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
#include <set>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    set<string> s;
    int people{ 0 }, cnt{ 1 };
    char lastWord = words[0][0];
    
    auto res = any_of(words.begin(), words.end(), [&](const auto& str)
    {
        if (++people > n) people = 1, ++cnt;

        if (s.find(str) != s.end() || lastWord != str.front())
            return true;
        
        s.insert(str);
        lastWord = str.back();
        return false;
    });
    
    return res ? vector<int>{ people, cnt } : vector<int>{ 0, 0 };
}